src/utils/include/utils/LCA.hpp

```diff
@@ -141,7 +143,7 @@ inline int LCA::queryRMQ( int i, int j ) const
 }
 
 // TODO: here is a suggestion, u and v should be unsigned! They are indexes...
-inline int LCA::getLCA( int u, int v ) const
+inline int LCA::getLCA( unsigned int u, unsigned int v ) const
 {
     assert( u >= 0 && u < m_firstAppearance.size() );
     assert( v >= 0 && v < m_firstAppearance.size() );
```

Comment:
Now that these are unsigned, you just need to test `assert( u < m_firstAppearance.size() && v < m_firstAppearance.size() );`, as they will be for sure greater or equal to zero